core/src/main/java/org/apache/iceberg/util/Tasks.java

```diff
@@ -300,49 +301,46 @@ private void tryRunOnFailure(I item, Exception failure) {
 
       for (final I item : items) {
         // submit a task for each item that will either run or abort the task
-        futures.add(service.submit(new Runnable() {
-          @Override
-          public void run() {
-            if (!(stopOnFailure && taskFailed.get())) {
-              // run the task with retries
-              boolean threw = true;
-              try {
-                runTaskWithRetry(task, item);
+        futures.add(service.submit(() -> {
+          if (!(stopOnFailure && taskFailed.get())) {
+            // run the task with retries
+            boolean threw = true;
+            try {
+              runTaskWithRetry(task, item);
```

Comment:
I don't think that suppressing warnings should change this class at all.

Reply:
revert